lua/cfc_ulx_commands/curse/cl_utils.lua

```diff
@@ -1,4 +1,85 @@
------ SETUP ----
+CFCUlxCurse = CFCUlxCurse or {}
+CFCUlxCurse._marchFolderCache = CFCUlxCurse._marchFolderCache or {}
+CFCUlxCurse._marchFolderCacheDroppedRoot = CFCUlxCurse._marchFolderCacheDroppedRoot or {}
+
+
+local marchFolderCache = CFCUlxCurse._marchFolderCache
+local marchFolderCacheDroppedRoot = CFCUlxCurse._marchFolderCacheDroppedRoot
+
+local fileFind = file.Find
+local tableInsert = table.insert
+local stringSub = string.sub
+
+
+--[[
+    - Returns a sequential list of all files in a folder, recursively.
+    - Includes the full file path and extension for each file.
+    
+    folder: (string)
+        - The folder to march. Exclude the trailing slash.
+        - sound, models, materials/models, etc.
+    path: (string)
+        - https://wiki.facepunch.com/gmod/File_Search_Paths
+        - GAME, DATA, etc.
+    out: (optional) (table)
+        - The table to append the file paths to.
+        - If nil, a new table will be created.
+
+    RETURNS: out (table)
+        - A table of file paths.
+--]]
```

Comment:
luals is your friend (and is what everyone uses)
```suggestion
--- Recursively gathers a list of all files in a given directory
--- (Includes the full file path and extension for each file)
---
--- @param folder string The folder to match, excluding the trailing slash (e.g. "sound", "models", etc.)
--- @param path string Search Path (e.g. "GAME", "DATA", etc.)
--- @param out? table The table to append the file paths to
--- @returns string[] out The collected file paths
```